docs/arcadia-users-group/20240730-arcadia-pycolor/lesson.md

```diff
@@ -0,0 +1,225 @@
+
+The package provides a function called `apc.mpl.setup` that sets the default `matplotlib` styles to match the Arcadia style guide. This function only needs to be called once, ideally at the beginning of the notebook or script. The styles it sets will automatically apply to all plots in the notebook or script.
+
+### What does `apc.mpl.setup` do?
+- Sets the default font to **"Suisse Int'l"** (or **"Arial"** if not installed)
+- Changes default axis, axis label, legend, and color cycler styles to match the Arcadia style
+- Registers Arcadia's colors, palettes, and gradients with `matplotlib`
+
+
+```python
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+
+apc.mpl.setup()
+
+def example_plot(title: str = "Example plot"):
+    plt.figure(figsize = (5, 3.5))
+    plt.plot([3, 1, 4, 1, 4], label = "group a")
+    plt.plot([2, 7, 1, 4, 2], label = "group b")
+
+    plt.xlabel("time")
+    plt.ylabel("value")
+
+    plt.legend(title = "category")
+    plt.title(title)
+
+# Plot using default rcParams
+with mpl.rc_context(mpl.rcParamsDefault):
+    example_plot("Using default rcParams")
+    plt.show()
+
+# Plot using apc.mpl.setup()
+example_plot("Using arcadia-pycolor")
+plt.show()
+```
+
+
+    
+![png](lesson_files/lesson_3_0.png)
+    
+
+
+
+    
+![png](lesson_files/lesson_3_1.png)
+    
+
+
+Some aspects of the style guide can only be applied to individual plots. The `apc.mpl.style_plot` function can be used to apply these styles to a single plot. This function takes a Matplotlib Axes object as input.
+
+If an `Axes` object is not passed to `style_plot`, the function will style the current plot (internally, `style_plot` uses `plt.gca()` to get the "current" `Axes` object).
+
+By default, the `style_plot` function capitalizes the x- and y-axis labels and styles the legend, if one exists. In addition, it has a few optional arguments that can be used to customize the styling of the x- and y-axis tick labels:
+
+- `monospaced_axes` sets the tick labels of the x- and/or y-axis to a monospaced font.
+- `categorical_axes` adjusts the x- and/or y-axis styles to be more readable when the axis represents a categorical variable.
+- `colorbar_exists` tells the function to style the colorbar, if one exists.
+
+
+```python
+example_plot("Using mpl.style_plot")
+apc.mpl.style_plot(monospaced_axes = "both")
+plt.show()
+```
+
+
+    
+![png](lesson_files/lesson_5_0.png)
+    
+
+
+## 3. Using Arcadia colors
+
+The Arcadia style guide defines sets of colors called "color palettes" that should be used in all figures. The `arcadia_pycolor` package provides easy access to both individual colors and to pre-defined palettes and gradients.
+
+Within `arcadia-pycolor`, all of Arcadia's named colors are available as variables. For example, you can access and preview the color `aegean` as follows. When the cell is evaluated, it will output the name and hex code of the color alongside a swatch showing what the color looks like.
```

Comment:
```suggestion
Within `arcadia-pycolor`, all of Arcadia's named colors are available as variables. For example, you can access and preview the color `aegean` as follows. When the cell is evaluated, it will output the name and hex code of the color alongside a swatch showing what the color looks like:
```